### ISSUE TITLE
Convnet Classifier (Debug?) Error

### ISSUE THREAD

**Issue by stlou** [ml4a/ml4a-ofx#25]
Trying to use the Convent Classifier app, and getting an error that I don't understand. Xcode lists it as a "Apple Mach-O Linker (Id) Error". 

Anyone seen this before?

At risk of overkill, I've pasted the error message here:

```
Undefined symbols for architecture x86_64:
  "GRT::Util::scale(double const&, double const&, double const&, double const&, double const&, bool)", referenced from:
      ofxGrtMatrixPlot::update(GRT::MatrixFloat const&, float, float) in ofxGrtMatrixPlot.o
  "GRT::ErrorLog::observerManager", referenced from:
      GRT::ErrorLog::triggerCallback(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const in ofxGrtTimeseriesPlot.o
  "GRT::ErrorLog::errorLoggingEnabled", referenced from:
      GRT::ErrorLog::ErrorLog(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in ofxGrtTimeseriesPlot.o
  "GRT::WarningLog::warningLoggingEnabled", referenced from:
      GRT::WarningLog::WarningLog(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in ofxGrtBarPlot.o
  "GRT::ClassificationData::addSample(unsigned int, GRT::VectorFloat const&)", referenced from:
      ofApp::update() in ofApp.o
  "GRT::VectorFloat::VectorFloat(GRT::VectorFloat const&)", referenced from:
      ofApp::update() in ofApp.o
  "GRT::WarningLog::observerManager", referenced from:
      GRT::WarningLog::triggerCallback(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const in ofxGrtBarPlot.o
  "GRT::KNN::~KNN()", referenced from:
      ofApp::setup() in ofApp.o
  "GRT::MLBase::predict(GRT::VectorFloat)", referenced from:
      ofApp::update() in ofApp.o
  "GRT::VectorFloat::~VectorFloat()", referenced from:
      ofApp::update() in ofApp.o
  "GRT::ClassificationData::clear()", referenced from:
      ofApp::clear() in ofApp.o
  "GRT::ClassificationData::load(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      ofApp::load() in ofApp.o
  "GRT::ClassificationData::save(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const", referenced from:
      ofApp::save() in ofApp.o
  "GRT::MLBase::train(GRT::ClassificationData)", referenced from:
      ofApp::trainClassifier() in ofApp.o
  "GRT::Log::baseLoggingEnabled", referenced from:
      GRT::Log const& GRT::Log::operator<<<char [74]>(char const (&) [74]) const in ofxGrtBarPlot.o
      GRT::Log::operator<<(std::__1::basic_ostream<char, std::__1::char_traits<char> >& (*)(std::__1::basic_ostream<char, std::__1::char_traits<char> >&)) const in ofxGrtBarPlot.o
      GRT::Log const& GRT::Log::operator<<<char [73]>(char const (&) [73]) const in ofxGrtBarPlot.o
      GRT::Log const& GRT::Log::operator<<<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const in ofxGrtTimeseriesPlot.o
      GRT::Log const& GRT::Log::operator<<<char [92]>(char const (&) [92]) const in ofxGrtTimeseriesPlot.o
      GRT::Log::operator<<(std::__1::basic_ostream<char, std::__1::char_traits<char> >& (*)(std::__1::basic_ostream<char, std::__1::char_traits<char> >&)) const in ofxGrtTimeseriesPlot.o
      GRT::Log const& GRT::Log::operator<<<char [81]>(char const (&) [81]) const in ofxGrtTimeseriesPlot.o
      ...
  "GRT::GestureRecognitionPipeline::getClassLikelihoods() const", referenced from:
      ofApp::update() in ofApp.o
  "GRT::GestureRecognitionPipeline::~GestureRecognitionPipeline()", referenced from:
      ofApp::ofApp() in main.o
      ofApp::~ofApp() in ofApp.o
  "GRT::ClassificationData::setNumDimensions(unsigned int)", referenced from:
      ofApp::setup() in ofApp.o
  "GRT::GestureRecognitionPipeline::setClassifier(GRT::Classifier const&)", referenced from:
      ofApp::setup() in ofApp.o
  "GRT::GestureRecognitionPipeline::GestureRecognitionPipeline()", referenced from:
      ofApp::ofApp() in main.o
  "GRT::MLBase::getTrained() const", referenced from:
      ofApp::draw() in ofApp.o
  "GRT::GestureRecognitionPipeline::getPredictedClassLabel() const", referenced from:
      ofApp::update() in ofApp.o
      ofApp::sendOSC() in ofApp.o
  "GRT::KNN::KNN(unsigned int, bool, bool, double, bool, unsigned int, unsigned int)", referenced from:
      ofApp::setup() in ofApp.o
  "GRT::GestureRecognitionPipeline::getNumClasses() const", referenced from:
      ofApp::trainClassifier() in ofApp.o
  "GRT::ClassificationData::ClassificationData(GRT::ClassificationData const&)", referenced from:
      ofApp::trainClassifier() in ofApp.o
  "GRT::VectorFloat::VectorFloat(unsigned long)", referenced from:
      ofApp::update() in ofApp.o
  "GRT::ClassificationData::ClassificationData(unsigned int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)", referenced from:
      ofApp::ofApp() in main.o
  "GRT::ClassificationData::~ClassificationData()", referenced from:
      ofApp::ofApp() in main.o
      ofApp::trainClassifier() in ofApp.o
      ofApp::~ofApp() in ofApp.o

ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by genekogan** [comment]
it's not finding GRT library. did you follow the instructions on [ofxGrt](https://github.com/nickgillian/ofxGrt)? most likely, it looks like you forgot to initialize the submodules.

**Comment by stlou** [comment]
Thanks for your response! Wasn't sure how much people were monitoring this, very much appreciate it. 

I believe I did follow the instructions for the ofxGrt addon. To make sure, I just went back and did an init and update on the submodule again. I was able to run the timeseries plot example, so it seems like the ofxGrt library is working fine, but maybe it's not correctly linked up with the ConvnetClassifier app?
